This patch allows to build speech tools honoring user provided CFLAGS,
CXXFLAGS and LDFLAGS. This is required to have a hardened build without
the obsolete hardening-wrapper package.

Author: Sergio Oller <sergioller@gmail.com>
Last update: 2014-08-02
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,97 @@
+dnl########################################################################
+dnl                                                                      ##
+dnl               Centre for Speech Technology Research                  ##
+dnl                    University of Edinburgh, UK                       ##
+dnl                      Copyright (c) 1996-2001                         ##
+dnl                       All Rights Reserved.                           ##
+dnl                                                                      ##
+dnl Permission is hereby granted, free of charge, to use and distribute  ##
+dnl this software and its documentation without restriction, including   ##
+dnl without limitation the rights to use, copy, modify, merge, publish,  ##
+dnl distribute, sublicense, and/or sell copies of this work, and to      ##
+dnl permit persons to whom this work is furnished to do so, subject to   ##
+dnl the following conditions:                                            ##
+dnl  1. The code must retain the above copyright notice, this list of    ##
+dnl     conditions and the following disclaimer.                         ##
+dnl  2. Any modifications must be clearly marked as such.                ##
+dnl  3. Original authors' names are not deleted.                         ##
+dnl  4. The authors' names are not used to endorse or promote products   ##
+dnl     derived from this software without specific prior written        ##
+dnl     permission.                                                      ##
+dnl                                                                      ##
+dnl THE UNIVERSITY OF EDINBURGH AND THE CONTRIBUTORS TO THIS WORK        ##
+dnl DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING      ##
+dnl ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT   ##
+dnl SHALL THE UNIVERSITY OF EDINBURGH NOR THE CONTRIBUTORS BE LIABLE     ##
+dnl FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES    ##
+dnl WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   ##
+dnl AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          ##
+dnl ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       ##
+dnl THIS SOFTWARE.                                                       ##
+dnl                                                                      ##
+dnl########################################################################
+AC_INIT(include/EST.h)
+
+AC_CANONICAL_SYSTEM
+AC_PROG_CXX
+AC_PROG_CC
+if test "x$GCC" = "xyes"; then
+	CFLAGS="$CFLAGS -Wall"
+fi
+AC_PROG_RANLIB
+AC_CHECK_TOOL(AR, ar)
+
+AC_C_BIGENDIAN
+
+AC_CHECK_LIB(ncurses, tputs, [TERMCAPLIB=-lncurses], [TERMCAPLIB=-lcurses])
+dnl if test "$TERMCAPLIB" != "-ltermcap"; then
+dnl    AC_CHECK_LIB(termcap, tputs, [TERMCAPLIB=-ltermcap], [TERMCAPLIB=-lncurses])
+dnl 
+AC_SUBST(TERMCAPLIB)
+
+dnl Which version of GCC do we have here
+COMPILERTYPE=gcc
+AC_TRY_COMPILE([#include <stdio.h>],
+               [int j= 
+                #if __GNUC__ == 2 && __GNUC_MINOR__ == 95
+                3;
+                #endif ],
+               [COMPILERTYPE=gcc295])
+AC_TRY_COMPILE([#include <stdio.h>],
+               [int j= 
+                #if __GNUC__ == 3 
+                3;
+                #endif ],
+               [COMPILERTYPE=gcc32])
+AC_TRY_COMPILE([#include <stdio.h>],
+               [ int j=
+                #if __GNUC__ == 2 && __GNUC_MINOR__ == 96
+                3;
+                #endif ],
+               [COMPILERTYPE=gcc296])
+AC_TRY_COMPILE([#include <stdio.h>],
+               [int j=
+                #if __GNUC__ == 2 && __GNUC_MINOR__ == 91
+                3;
+                #endif ],
+               [COMPILERTYPE=egcs])
+AC_TRY_COMPILE([#include <stdio.h>],
+               [int j=
+                #if __GNUC__ == 2 && __GNUC_MINOR__ == 8
+                3;
+                #endif ],
+               [COMPILERTYPE=gcc28])
+AC_SUBST(COMPILERTYPE)
+
+LINUXAUDIO=none
+AC_TRY_COMPILE([#include <alsa/version.h>],
+               [int j=
+                #if SND_LIB_SUBMINOR >= 11
+                3;
+                #endif],
+              [LINUXAUDIO="alsa"])
+AC_SUBST(LINUXAUDIO)
+
+AC_CONFIG_FILES([config/config config/user_flags.mak])
+
+AC_OUTPUT
--- a/configure.in
+++ /dev/null
@@ -1,94 +0,0 @@
-dnl########################################################################
-dnl                                                                      ##
-dnl               Centre for Speech Technology Research                  ##
-dnl                    University of Edinburgh, UK                       ##
-dnl                      Copyright (c) 1996-2001                         ##
-dnl                       All Rights Reserved.                           ##
-dnl                                                                      ##
-dnl Permission is hereby granted, free of charge, to use and distribute  ##
-dnl this software and its documentation without restriction, including   ##
-dnl without limitation the rights to use, copy, modify, merge, publish,  ##
-dnl distribute, sublicense, and/or sell copies of this work, and to      ##
-dnl permit persons to whom this work is furnished to do so, subject to   ##
-dnl the following conditions:                                            ##
-dnl  1. The code must retain the above copyright notice, this list of    ##
-dnl     conditions and the following disclaimer.                         ##
-dnl  2. Any modifications must be clearly marked as such.                ##
-dnl  3. Original authors' names are not deleted.                         ##
-dnl  4. The authors' names are not used to endorse or promote products   ##
-dnl     derived from this software without specific prior written        ##
-dnl     permission.                                                      ##
-dnl                                                                      ##
-dnl THE UNIVERSITY OF EDINBURGH AND THE CONTRIBUTORS TO THIS WORK        ##
-dnl DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING      ##
-dnl ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT   ##
-dnl SHALL THE UNIVERSITY OF EDINBURGH NOR THE CONTRIBUTORS BE LIABLE     ##
-dnl FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES    ##
-dnl WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   ##
-dnl AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          ##
-dnl ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       ##
-dnl THIS SOFTWARE.                                                       ##
-dnl                                                                      ##
-dnl########################################################################
-AC_INIT(include/EST.h)
-
-AC_CANONICAL_SYSTEM
-AC_PROG_CC
-if test "x$GCC" = "xyes"; then
-	CFLAGS="$CFLAGS -Wall"
-fi
-AC_PROG_RANLIB
-AC_CHECK_TOOL(AR, ar)
-
-AC_C_BIGENDIAN
-
-AC_CHECK_LIB(ncurses, tputs, [TERMCAPLIB=-lncurses], [TERMCAPLIB=-lcurses])
-dnl if test "$TERMCAPLIB" != "-ltermcap"; then
-dnl    AC_CHECK_LIB(termcap, tputs, [TERMCAPLIB=-ltermcap], [TERMCAPLIB=-lncurses])
-dnl 
-AC_SUBST(TERMCAPLIB)
-
-dnl Which version of GCC do we have here
-COMPILERTYPE=gcc
-AC_TRY_COMPILE([#include <stdio.h>],
-               [int j= 
-                #if __GNUC__ == 2 && __GNUC_MINOR__ == 95
-                3;
-                #endif ],
-               [COMPILERTYPE=gcc295])
-AC_TRY_COMPILE([#include <stdio.h>],
-               [int j= 
-                #if __GNUC__ == 3 
-                3;
-                #endif ],
-               [COMPILERTYPE=gcc32])
-AC_TRY_COMPILE([#include <stdio.h>],
-               [ int j=
-                #if __GNUC__ == 2 && __GNUC_MINOR__ == 96
-                3;
-                #endif ],
-               [COMPILERTYPE=gcc296])
-AC_TRY_COMPILE([#include <stdio.h>],
-               [int j=
-                #if __GNUC__ == 2 && __GNUC_MINOR__ == 91
-                3;
-                #endif ],
-               [COMPILERTYPE=egcs])
-AC_TRY_COMPILE([#include <stdio.h>],
-               [int j=
-                #if __GNUC__ == 2 && __GNUC_MINOR__ == 8
-                3;
-                #endif ],
-               [COMPILERTYPE=gcc28])
-AC_SUBST(COMPILERTYPE)
-
-LINUXAUDIO=none
-AC_TRY_COMPILE([#include <alsa/version.h>],
-               [int j=
-                #if SND_LIB_SUBMINOR >= 11
-                3;
-                #endif],
-              [LINUXAUDIO="alsa"])
-AC_SUBST(LINUXAUDIO)
-
-AC_OUTPUT(config/config)
--- /dev/null
+++ b/config/user_flags.mak.in
@@ -0,0 +1,45 @@
+ ###########################################################################
+ ##                                                                       ##
+ ##                Centre for Speech Technology Research                  ##
+ ##                     University of Edinburgh, UK                       ##
+ ##                         Copyright (c) 1996                            ##
+ ##                        All Rights Reserved.                           ##
+ ##                                                                       ##
+ ##  Permission is hereby granted, free of charge, to use and distribute  ##
+ ##  this software and its documentation without restriction, including   ##
+ ##  without limitation the rights to use, copy, modify, merge, publish,  ##
+ ##  distribute, sublicense, and/or sell copies of this work, and to      ##
+ ##  permit persons to whom this work is furnished to do so, subject to   ##
+ ##  the following conditions:                                            ##
+ ##   1. The code must retain the above copyright notice, this list of    ##
+ ##      conditions and the following disclaimer.                         ##
+ ##   2. Any modifications must be clearly marked as such.                ##
+ ##   3. Original authors' names are not deleted.                         ##
+ ##   4. The authors' names are not used to endorse or promote products   ##
+ ##      derived from this software without specific prior written        ##
+ ##      permission.                                                      ##
+ ##                                                                       ##
+ ##  THE UNIVERSITY OF EDINBURGH AND THE CONTRIBUTORS TO THIS WORK        ##
+ ##  DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING      ##
+ ##  ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT   ##
+ ##  SHALL THE UNIVERSITY OF EDINBURGH NOR THE CONTRIBUTORS BE LIABLE     ##
+ ##  FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES    ##
+ ##  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   ##
+ ##  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          ##
+ ##  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       ##
+ ##  THIS SOFTWARE.                                                       ##
+ ##                                                                       ##
+ ###########################################################################
+ ##                                                                       ##
+ ##                 Author: Sergio Oller <sergioller@gmail.com>           ##
+ ##                   Date: Jan 6 2014                                    ##
+ ## --------------------------------------------------------------------  ##
+ ## Settings for compiler defined via autoconf                            ##
+ ##                                                                       ##
+ ###########################################################################
+
+USER_CFLAGS    = @CFLAGS@
+USER_CXXFLAGS  = @CXXFLAGS@
+USER_LINKFLAGS = @LDFLAGS@
+
+
--- a/config/Makefile
+++ b/config/Makefile
@@ -40,7 +40,7 @@
 FILES = Makefile ReadMe example.Makefile example.module.mak\
 	 common_make_rules test_make_rules vc_common_make_rules  \
 	config.in vc_config_make_rules-dist \
-	project.mak system.sh make_system.mak
+	project.mak system.sh make_system.mak user_flags.mak
 
 #LOCAL_CLEAN = modinclude*.inc system.mak
 
--- a/config/config.in
+++ b/config/config.in
@@ -200,6 +200,8 @@
 
 include $(EST)/config/compilers/$(COMPILER).mak
 
+include $(EST)/config/user_flags.mak
+
 ifneq ($(JAVA_COMPILER),none)
     include $(EST)/config/compilers/$(JAVA_COMPILER).mak
 endif
--- a/config/rules/defaults.mak
+++ b/config/rules/defaults.mak
@@ -164,13 +164,13 @@
 ## Default C Compilation
 
 ifndef	CC_COMMAND
-    CC_COMMAND = $(CC) -c $(CFLAGS) $(COMPILE_CCFLAGS) $(DEFINES) $(INCLUDES)
+    CC_COMMAND = $(CC) -c $(CFLAGS) $(COMPILE_CCFLAGS) $(DEFINES) $(INCLUDES) $(USER_CFLAGS)
 endif
 
 ## Default C++ Compilation
 
 ifndef CXX_COMMAND
-    CXX_COMMAND = $(CXX) -c $(CXXFLAGS) $(COMPILE_CXXFLAGS) $(DEFINES) $(INCLUDES) $(TEMPLATES) 
+    CXX_COMMAND = $(CXX) -c $(CXXFLAGS) $(COMPILE_CXXFLAGS) $(DEFINES) $(INCLUDES) $(TEMPLATES) $(USER_CXXFLAGS)
 endif
 
 ## C++ for dynamic loading
@@ -196,6 +196,6 @@
 ## Link a program (not including libraries)
 
 ifndef LINK_COMMAND
-    LINK_COMMAND = $(CXX) $(LINKFLAGS) $(TEMPLATES) 
+    LINK_COMMAND = $(CXX) $(LINKFLAGS) $(TEMPLATES) $(USER_LINKFLAGS)
 endif
 
--- a/config/compilers/gcc_defaults.mak
+++ b/config/compilers/gcc_defaults.mak
@@ -82,7 +82,7 @@
 SHARED_LINKFLAGS = 
 
 ifndef GCC_MAKE_SHARED_LIB
-    MAKE_SHARED_LIB = $(CXX) -shared -o XXX -Wl,-soname -Wl,YYY
+    MAKE_SHARED_LIB = $(CXX) -shared -o XXX -Wl,-soname -Wl,YYY $(USER_LINKFLAGS)
 else
     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
 endif
